Nitpicks

gperciva opened the pull request.

cperciva commented on imds-filterd/ident.c:
```diff
@@ -54,6 +54,7 @@ gotdata(void * cookie, ssize_t len)
 	struct cstate * cs = cookie;
 	struct sockaddr_in addrs[2];
 	struct xucred uc;
+	int alloclen;
```
Why "alloclen"?  Seems like "printlen" would make more sense?

gperciva replied:
No particular reason.  I mean, first I went `len`, then figured that wasn't specific enough, then I thought `strlen`, then I realized that it would be insane to copy a function name from the standard library.  I'll switch it to `printlen`.

gperciva committed:
Partial libcperciva import

This only updates the files that are necessary in order to silence
implicit casting warnings in these files.

gperciva committed:
Prevent "possibly uninitialized variable" warnings

Checking the `best_rtmsk` variable (on line 189) ensures that these
variables were also initialized, but both gcc and clang fail to
recognize that.

gperciva committed:
Reassure clang-scan that we're not using this 'linelen'

Without this, clang-scan complains that:

----
request.c:113:7: warning: Although the value stored to 'linelen' is used in the enclosing expression, the value is never actually read from 'linelen' [deadcode.DeadStores]
        if ((linelen = getline(&line, &linecap, f)) <= 0) {
             ^         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
----

It's useful to keep the `linelen = getline(...)` format, because we use
it approximately 50 lines lower, within a while() loop.

gperciva committed:
Check return value of sprintf

gperciva commented:
Updated, and tweaked the comment in 61823ed.
